email-native/src/main/java/org/ballerinalang/stdlib/email/server/EmailListener.java

```diff
@@ -64,25 +67,50 @@ public boolean onMessage(EmailEvent emailEvent) {
         if (runtime != null) {
             Set<Map.Entry<String, ObjectValue>> services = registeredServices.entrySet();
             for (Map.Entry<String, ObjectValue> service : services) {
-                runtime.invokeMethodSync(service.getValue(), ON_MESSAGE, null, ON_MESSAGE_METADATA, email, true);
+                invokeMethodSync(service.getValue(), ON_MESSAGE, ON_MESSAGE_METADATA, email, true);
             }
         } else {
             log.error("Runtime should not be null.");
         }
         return true;
     }
 
+    private void invokeMethodSync(ObjectValue object, String methodName, StrandMetadata metadata, Object... args) {
+        Semaphore semaphore = new Semaphore(0);
```

Comment:
Declaring a new Semaphore in each method call may not limit the access. May be we can declare it only once.

Reply:
This method is to replace the removed function here: https://github.com/ballerina-platform/ballerina-lang/pull/25744/files#diff-a3b2c474ab6a7c3e36374099c3bf5e9eL208-L233